File: backend/inventory-service/src/main/java/com/enterprise/ecommerce/inventory/service/InventoryService.java
package com.enterprise.ecommerce.inventory.service;

import com.enterprise.ecommerce.inventory.dto.*;

public interface InventoryService {

    InventoryResponse getByProductId(Long productId);

    CheckAvailabilityResponse checkAvailability(CheckAvailabilityRequest request);

    InventoryResponse reserve(InventoryOperationRequest request);

    InventoryResponse confirm(InventoryOperationRequest request);

    InventoryResponse release(InventoryOperationRequest request);

    InventoryResponse restore(InventoryOperationRequest request);

    InventoryResponse updateStock(Long productId, UpdateInventoryRequest request);
}

File: frontend/src/services/inventoryService.js
import axiosClient from '../api/axiosClient';

const unwrap = (response) => response.data.data;

export const inventoryService = {
  getByProductId: async (productId) => {
    const response = await axiosClient.get(`/api/inventory/${productId}`);
    return unwrap(response);
  },

  updateStock: async (productId, availableQuantity) => {
    const response = await axiosClient.put(`/api/inventory/${productId}`, {
      availableQuantity,
    });
    return unwrap(response);
  },
};
